src/clj_commons/byte_streams.clj

```diff
@@ -429,17 +429,17 @@
 
 ;; byte-buffer => sequence of byte-buffers
 (def-conversion ^{:cost 0} [ByteBuffer (vector-of ByteBuffer)]
-  [buf {:keys [chunk-size]}]
+  [buf {:keys [^long chunk-size]}]
   (if chunk-size
     (let [lim (.limit buf)
-          indices (range (.position buf) lim ^long chunk-size)]
+          indices (range (.position buf) lim chunk-size)]
       (mapv
-        #(-> buf
-           .duplicate
-           ^ByteBuffer (.position (int %))
-           ^ByteBuffer (.limit (int (min lim (+ (int %) ^long chunk-size))))
-           .slice)
-        indices))
+       #(-> buf
+            .duplicate
+            ^ByteBuffer (.position (int %))
+            ^ByteBuffer (.limit (int (min lim (p/+ (int %) chunk-size))))
```

Comment:
Hmmm. I wonder why we downcast the index to an int. Since chunk-size is long, the index will be immediately widened back to a long for the addition.